ci_framework/roles/env_op_images/tasks/main.yml

```diff
@@ -0,0 +1,111 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Ensure directory is present
+  ansible.builtin.file:
+    path: "{{ cifmw_env_op_images_dir }}/{{ item }}"
+    state: directory
+  loop:
+    - artifacts
+    - logs
+
+- name: Check if OpenStackControlPlane is setup
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.shell: |
+    set -o pipefail &&
+    oc get OpenStackControlPlane -n {{ cifmw_install_yamls_defaults['NAMESPACE'] }} --no-headers=true | awk '{print $2}'
+  register: openstack_installed
+
+- name: Get openstack operator images
+  when:
+    - openstack_installed.stdout == "True"
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  block:
+    - name: Create a file to store the images
+      ansible.builtin.copy:
+        dest: "{{ cifmw_env_op_images_dir }}/artifacts/{{ cifmw_env_op_images_file  }}"
+        content: |
+          cifmw_install_yamls_vars:
+          cifmw_openstack_operator_images:
+          cifmw_edpm_prepare_extra_vars:
+
+    - name: Get images from the csv
+      ansible.builtin.shell: |
+        oc get csv -n {{ cifmw_install_yamls_defaults['OPERATOR_NAMESPACE'] }} openstack-operator.v0.0.1 -o=jsonpath='{.spec.install.spec.deployments[0].spec.template.spec.containers[1].env[*]}'
+      register: csv_images
+
+    - name: Format the output
+      ansible.builtin.set_fact:
+        csv_env_images: "{{ csv_images.stdout| regex_findall('\\{[^}]+\\}') }}"
+
+    - name: Extract env variable name and images
+      loop: "{{ csv_env_images }}"
+      ansible.builtin.set_fact:
+        csv_image_list: "{{ csv_image_list | default([]) + [item | from_json] }}"
+
+    - name: Write images to file
+      loop: "{{ csv_image_list }}"
+      ansible.builtin.lineinfile:
+        path: "{{ cifmw_env_op_images_dir }}/artifacts/{{ cifmw_env_op_images_file  }}"
+        line: "  {{item.name}}: {{item.value}}"
+        insertafter: "cifmw_edpm_prepare_extra_vars:"
+
+    - name: Get all the pods in openstack-operator namespace
```

Comment:
I see that you trust that the block KUBECONFIG provides auth info, that can do the job, but  I'd prefer to be consistent all across the framework. Can you please check how we use this module?

Reply:
Added the auth parameters